include/tvm/meta_schedule/database.h

```diff
@@ -168,8 +176,16 @@ class TuningRecord : public runtime::ObjectRef {
 /* \brief The abstract interface of database. */
 class DatabaseNode : public runtime::Object {
  public:
+  /*!
+   * \brief Constructor
+   * \param mod_eq_name A string to specify the module equality testing and hashing method.
+   *  It must be one of the followings:
+   *    - "structural": Use StructuralEqual/Hash
+   */
+  explicit DatabaseNode(String mod_eq_name = "structural");
+
   /*! \brief Default destructor */
-  virtual ~DatabaseNode() = default;
```

Comment:
May I ask why this is removed to `database.cc` instead?

Reply:
This is necessary to be able to hold `std::unique_ptr<ModuleEquality>` in `DatabaseNode` where `ModuleEquality` is forward declared. You can google "unique_ptr incomplete type".